### DISCUSSION TITLE
The extension 'SQLTools PostgreSQL/Cockroach Driver' wants to sign in using SQLTools Driver Credentials.

### DISCUSSION THREAD

**Discussion by Yahud100** [mtxr/vscode-sqltools#1409] [Q&A] [answered]
While programmatically connecting to the database using the method:

conn = await this._setConnection(conn as IConnection).catch(e => this.errorHandler('Error opening connection', e));

a pop-up frequently appears:
'The extension "SQLTools PostgreSQL/Cockroach Driver" wants to sign in using SQLTools Driver Credentials.'

After allowing, another prompt requests the password:
"Optionally use the key button to store the password securely until you sign out using the 'Accounts' menu."

Although the connection to the database succeeds, this behavior is inconsistent.
Key Questions to Address:

1) What triggers the pop-up sporadically despite prior connections?
2) How can this behavior be disabled for smoother automated connections?
3) Is it related to credential storage or session handling issues?
Clear insights and resolution steps are required to eliminate these disruptions.

**Comment by gjsjohnmurray** [comment, marked as answer]
> While programmatically connecting to the database using the method:
> 
> conn = await this._setConnection(conn as IConnection).catch(e => this.errorHandler('Error opening connection', e));

I'm not clear whether you are doing this is code of your own, or if your comment refers to this line in the source of the SQLTools extension:

https://github.com/mtxr/vscode-sqltools/blob/91fa2b9db752a3d98cb1c3d3257baa285264bf2b/packages/plugins/connection-manager/extension.ts#L205

> a pop-up frequently appears:

I assume when you use the Edit Connection option from the context menu of your connection definition in the SQLTools Connections tree. then the **Use password** field is set to **SQLTools Driver Credentials**

Does your VS Code Accounts menu (from the person icon near the bottom of Activity Bar) show any entries ending in `(SQLTools Driver Credentials)`?

The first popup is informing you that the PostgeSQL/Cockroach driver wants to use the authentication provider that the main SQLTools extension implements.

When you enter your password at the second prompt are you clicking the key icon in order to save the password? Or are you simply pressing Enter?

If you do not use the key icon at this point then I would expect you to get both prompts again if you close/reopen your VS Code window, or run the `Developer: Reload Window` command from Command Palette (F1).

If you **do** use the key icon I wouldn't expect either prompt to reappear unless you use the Sign Out option from the corresponding entry on the Accounts menu of VS Code.

**Comment by Yahud100** [comment]
thanks for support , i get required insights from this conversation
